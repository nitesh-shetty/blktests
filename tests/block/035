#!/bin/bash
# SPDX-License-Identifier: GPL-3.0+
# Copyright (C) 2023 Nitesh Shetty
#
# Run a single copy offload IO using device copy offload on a null block device

. tests/block/rc
. common/null_blk

DESCRIPTION="run single copy offload IO using device copy offload on null-blk"

requires() {
	_have_fio && _have_tracefs &&
	_have_null_blk && _have_module_param null_blk copy_max_bytes && _have_fio
}

test () {
	echo "Running ${TEST_NAME}"

	if ! _init_null_blk memory_backed=1; then
		return 1
	fi
	if ! _have_tracepoint "nullb/nullb_copy_op"; then
		return 1
	fi
	_fio_copy_verify_single_io "/dev/nullb0" 4096 "nullb/nullb_copy_op"

	_exit_null_blk

	echo "Test complete"
}
