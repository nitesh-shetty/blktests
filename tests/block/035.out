Running block/035
 fio-XXXXX [XXX] .....    X.XXXXXX: nullb_copy_op: disk=X, req=copy : dst=4096, src=0, len=X
Test complete
